### PR TITLE
Update documentation to use new operation.with_directives decorator

### PR DESCRIPTION
**Pull request by b-butler**
## Description

This changes the documentation to use the `FlowProject.operation.with_directives` decorator over `flow.directives`. This is to teach a more coherent syntax with `FlowGroupEntry.with_directives`.

## Motivation and Context

Is the corresponding documentation PR for https://github.com/glotzerlab/signac-flow/pull/502.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-docs/blob/master/ContributorAgreement.md).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md#code-style) of this project.

### CODE REVIEW

**Comment by b-butler** [comment]
@atravitz and @Tobias-Dwyer will you guys have time to look at this sometime next week? We just merged https://github.com/glotzerlab/signac-flow/pull/502 and this goes with that.

**Review by kidrahahjo — APPROVED** [review]
The changes look good, I checked docs.signac.io and saw that wall time and memory directives are missing. I'll make a PR that adds them.

File: docs/source/cluster_submission.rst
Comment (bdice) [inline docs/source/cluster_submission.rst:114]:
```suggestion
    Note that all directives may be specified as callables, e.g. ``FlowProject.operation.with_directives({"np": lambda job: job.doc.np})``.
```

File: docs/source/flow-group.rst
Comment (bdice) [inline docs/source/flow-group.rst:93]: Make the style of dict syntax match.
```suggestion
    @ex.with_directives({"ngpu": 2})
```

File: docs/source/recipes.rst
Comment (bdice) [inline docs/source/recipes.rst:269]: Needs rewritten.
```suggestion
The ``@FlowProject.operation.with_directives(...)`` decorator provides additional instructions on how to execute this operation.
The decorator ``@FlowProject.operation`` does not assign any directives to the operation.
```

File: docs/source/recipes.rst
Comment (bdice) [inline docs/source/recipes.rst:329]: It's weird to mix single and double quotes.
```suggestion
    @Project.operation.with_directives({"executable": "singularity exec software.simg python"})
```

**Review by bdice — CHANGES_REQUESTED** [review]

File: docs/source/recipes.rst
Comment (bdice) [inline docs/source/recipes.rst:266]: Is this example still useful, or should we recommend the user to use the `nranks` directive and adapt the command to remove `mpiexec`?

File: docs/source/recipes.rst
Comment (bdice) [inline docs/source/recipes.rst:270]: Wherever possible, we should link to the corresponding API docs, e.g. with
```suggestion
The decorator :py:meth:`FlowProject.operation` does not assign any directives to the operation.
```

**Review by bdice — CHANGES_REQUESTED** [review]

**Comment by kidrahahjo** [comment]
@bdice #134 is ready to merge once @vyasr or @klywang approves it.

File: docs/source/recipes.rst
Comment (b-butler) [reply docs/source/recipes.rst:266]: This whole section on MPI is out of date and actually would likely not work given the `submit -> run -> exec` model. However, I don't know if it is in scope to modify this in this PR.

File: docs/source/recipes.rst
Comment (b-butler) [reply docs/source/recipes.rst:270]: This needs to be mostly reworked in another PR to fix the out of data MPI documentation, so let's create an issue and fix that there.

**Comment by b-butler** [comment]
@bdice I made an issue #144 to document the needed changes to the MPI documentation in recipes. I think these changes should go in a separate PR. If you agree can you provide another review, if not then let me know, so I can make the changes here.

File: docs/source/recipes.rst
Comment (bdice) [reply docs/source/recipes.rst:270]: I might not have been clear enough. This comment applies to a lot of places not related to the MPI docs. I'll mark them.

**Review by bdice — APPROVED** [review]

**Pull request merged**
